service/k8sclient/k8sclient.go

```diff
@@ -1319,45 +1342,40 @@ func (c *K8sClient) CheckOperators(ctx context.Context) (*Operators, error) {
 	apiVersions := strings.Split(string(output), "\n")
 
 	return &Operators{
-		Xtradb: c.checkOperatorStatus(apiVersions, pxcAPIVersion),
-		Psmdb:  c.checkOperatorStatus(apiVersions, psmdbAPIVersion),
+		Xtradb: c.checkOperatorStatus(apiVersions, pxcAPINamespace),
+		Psmdb:  c.checkOperatorStatus(apiVersions, psmdbAPINamespace),
 	}, nil
 }
 
 // checkOperatorStatus returns if operator is installed and operators version.
 // It checks for all API versions supported by the operator and based on the latest API version in the list
 // figures out which version of operator is installed.
-func (c *K8sClient) checkOperatorStatus(installedVersions []string, expectedAPIVersion string) (operator Operator) {
-	apiNamespace := strings.Split(expectedAPIVersion, "/")[0]
+func (c *K8sClient) checkOperatorStatus(installedVersions []string, apiPrefix string) (operator Operator) {
 	operator.Status = OperatorStatusNotInstalled
 	lastVersion, _ := version.NewVersion("v0.0.0")
+	zeroVersion := lastVersion
 	for _, apiVersion := range installedVersions {
-		if !strings.HasPrefix(apiVersion, apiNamespace) {
+		if !strings.HasPrefix(apiVersion, apiPrefix) {
 			continue
 		}
-		if apiVersion == expectedAPIVersion {
-			operator.Status = OperatorStatusOK
-		}
-		if operator.Status == OperatorStatusNotInstalled {
-			operator.Status = OperatorStatusUnsupported
-		}
 		v := strings.Split(apiVersion, "/")[1]
-
 		versionParts := strings.Split(v, "-")
 		if len(versionParts) != 3 {
 			continue
 		}
 		v = strings.Join(versionParts, ".")
 		newVersion, err := version.NewVersion(v)
 		if err != nil {
-			c.l.Warn("can't parse version %s: %s", v, err)
+			c.l.Warnf("can't parse version %s: %s", v, err)
 			continue
 		}
-		if newVersion.LessThanOrEqual(lastVersion) {
-			continue
+		if newVersion.GreaterThan(lastVersion) {
+			lastVersion = newVersion
 		}
-		lastVersion = newVersion
-		operator.Version = lastVersion.String()
+	}
+	operator.Version = lastVersion.String()
+	if lastVersion != zeroVersion { // comparing pointers
+		operator.Status = OperatorStatusOK
 	}
```

Comment:
we should check if operator version exist in version service and return ok if it is, or unsupported if it's not.

Reply:
Is it good idea to request version service on two places like that?

Reply:
Then we should move it from here to pmm-managed or do opposite.

Reply:
I mean having two places where you have to maintain the abstractions.